package kite;

message Xyz {
  required double x = 1;
  required double y = 2;
  required double z = 3;
}

message Dcm {
  required double r11 = 1;
  required double r12 = 2;
  required double r13 = 3;

  required double r21 = 4;
  required double r22 = 5;
  required double r23 = 6;

  required double r31 = 7;
  required double r32 = 8;
  required double r33 = 9;
}

message CarouselState {
  required Xyz kiteXyz = 1;
  required Dcm kiteDcm = 2;
  required double delta = 3;
  required double rArm = 4;
  required double zt = 5;
  repeated string messages = 6;
  required double w0 = 7;
}

message PendulumOpt {
  repeated double x = 1;
  repeated double z = 2;
  //  repeated string messages = 3;
  required double endTime = 3;
  required int32 iters = 4;
}

message KiteOpt {
  repeated CarouselState css = 1;
  required double endTime = 2;
  required int32 iters = 3;
  required double wind_x = 4;
}
